src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/ISyntaxFactsService.cs

```diff
@@ -360,6 +360,8 @@ internal interface ISyntaxFactsService : ILanguageService
         string GetDisplayName(SyntaxNode node, DisplayNameOptions options, string rootNamespace = null);
 
         SyntaxNode GetContainingTypeDeclaration(SyntaxNode root, int position);
+        string GetNameOfContainingType(SyntaxNode node);
```

Comment:
Instead of adding a new API, I think a better approach is for the client to invoke the existing syntax fact APIs `GetContainingTypeDeclaration` and `GetContainingMemberDeclaration`, and if non-null node is returned, then invoke `GetSymbolInfo` on it to get the containing type/member symbol.